### PR TITLE
Updated AI tags and proximity reloading and DangerTracker

### PR DESCRIPTION
**Pull request by kbatbouta**
## Additions

- Colonists can now use  allowed ammo  around them when they are out and do not have any sidearms. The ammo is selected from a space of 3x3 cells around the colonist (unless they have an AOE weapon they will just revert to the old system).
- Added flare gun and smoke grenades to most industrial raiders at small numbers and grenade launchers to advance  raiders/`PawnKind` also in very few numbers. (ranging from a 0.05 chance to 0.1 chance and 1 to 3 rounds)
- Fixed a weapons tag issue in Xml.
- Added `ammoTags` to AmmoDef.
- Added `DangerTracker` which is used for helping pawns avoid getting shot in the back by allies and to improve the path finding by avoid being so close to the shooters.

## Changes

- Synced the AI code with the most recent AI tags.

### CODE REVIEW

File: Source/CombatExtended/CombatExtended/Defs/AmmoDef.cs [hunk 1]
```diff
@@ -22,6 +22,8 @@ public class AmmoDef : ThingDef
         // mortar ammo should still availabe when the ammo system is off
         public bool isMortarAmmo = false;
 
+        public List<string> ammoTags;
```
Comment (Alicecomma) [inline Source/CombatExtended/CombatExtended/Defs/AmmoDef.cs:25]: unused?

Comment (kbatbouta) [reply Source/CombatExtended/CombatExtended/Defs/AmmoDef.cs:25]: no, so the team can start working on the ammo tag system, if it's not populated i can do anything with it thus this is the first step.

File: Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs [hunk 2]
```diff
@@ -906,6 +915,8 @@ public override void Tick()
                     }
                 }
             }
+            if (shotHeight < 2.0f && originInt.DistanceTo(positionInt) > 3)
```
Comment (Alicecomma) [inline Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs]: reference to CollisionVertical.WallCollisionHeight would be good

Comment (kbatbouta) [reply Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs]: will do

**Review by Alicecomma — APPROVED** [review]
lgtm

**Review by perkinslr — APPROVED** [review]
I would probably have added and used `DistanceToSegmentSq`, instead of `DistanceToSegment`, but this code path shouldn't be hot enough to worry about it.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**